Index: sdi-entrega2-2021-501-2021-504/routes/rapiusuarios.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/sdi-entrega2-2021-501-2021-504/routes/rapiusuarios.js b/sdi-entrega2-2021-501-2021-504/routes/rapiusuarios.js
new file mode 100644
--- /dev/null	(date 1620394266663)
+++ b/sdi-entrega2-2021-501-2021-504/routes/rapiusuarios.js	(date 1620394266663)
@@ -0,0 +1,34 @@
+module.exports = function(app,gestorBD){
+
+    app.post("/api/autenticar/",function (req,res) {
+        let seguro = app.get("crypto").createHmac('sha256',app.get('clave'))
+            .update(req.body.password).digest('hex');
+
+        let criterio ={
+            email:req.body.email,
+            password:seguro
+        }
+        gestorBD.obtenerUsuarios(criterio,function (usuarios) {
+            if(usuarios==null || usuarios.length==0){
+                res.status(401); //unauthorized
+                res.json({
+                    autenticado:false
+                })
+            }else{
+                let token = app.get('jwt').sign(
+                    {usuario: criterio.email , tiempo: Date.now()/1000},
+                    "secreto");
+                req.session.usuario = criterio.email;
+                req.session.rol=usarios[0].rol;
+                if(req.session.rol == "usuario"){
+                    req.session.money=usuarios[0].money;
+                }
+                res.status(200);
+                res.json({
+                    autenticado:true,
+                    token:token
+                })
+            }
+        })
+    })
+}
\ No newline at end of file
